src/components/contentSection/ContentSection.jsx

```diff
@@ -0,0 +1,194 @@
+import "./contentSection.styles.css";
+import Cart from "../cart";
+import Content from "../content";
+import Sidebar from "../sidebar";
+import menuList from "../../models/menuModel";
+import { getDishByMenu } from "../../models/dishModel";
+import { useState, useEffect } from "react";
+
+export const ContentSection = ({
+  activeMenuIndex,
+  isVegOnly,
+  searchKeyword,
+  setActiveMenuIndex,
+}) => {
+  const [activeMenuItems, setActiveMenuItems] = useState([]);
+  const [cartItems, setCartItems] = useState([]);
+  const [checkoutMessage, setCheckoutMessage] = useState("");
+  const [cartEmptyMessage, setCartEmptyMessage] = useState("");
+
+  useEffect(() => {
+    onSidebarClick(activeMenuIndex);
+  }, [isVegOnly, searchKeyword, cartItems]);
+
+  const onSidebarClick = (index) => {
+    let updatedMenuItems = getDishByMenu(menuList[index]);
+    if (searchKeyword !== "") {
+      updatedMenuItems = updatedMenuItems.filter((item) =>
+        item.name.toLowerCase().includes(searchKeyword.trim().toLowerCase())
+      );
+    }
+    for (let i = 0; i < updatedMenuItems.length; i++) {
```

Comment:
try to avoid for loops, there is almost always a way to use map/foreach/reduce instead.

Reply:
for loop is replaced with map.